data/convert_to_triples.R

```diff
@@ -0,0 +1,442 @@
+triples_df <- bind_rows(
+  reformatted_ontology,
+  reformatted_references,
+  reformatted_reviewers,
+  reformatted_units,
+  reformatted_hierarchy,
+  reformatted_categorical,
+  reformatted_traits
+)
+
+#remove NA's; remove stray numbers required before to create unique column names 
+triples_df <- triples_df %>% 
+  filter(!is.na(Object)) %>%
+  mutate(Predicate = stringr::str_replace(Predicate, "\\>[:digit:]", "\\>"))
+
+
+# rdflib can't handle UTF-8 :(, 
+# We can either "transliterate" our UTF-8 to ASCII (i.e. drop accent marks)
+# or we can replace with Unicode, which we can later un-encode back to the original UTF-8
+triples_df %>% 
+  mutate(Object = iconv(Object, from="UTF-8", to="ASCII", sub="Unicode")) %>%
+#  mutate(Object = iconv(Object, from="UTF-8", to="ASCII/TRANSLIT")) %>%
+  mutate(graph = ".") %>% # quads have a fourth column, usually "."
+  filter(Object != "<NA>", Subject != "<NA>", Predicate != "<NA>") %>% # we have some NAs sneaking in as URIs
+  mutate(Object = gsub("\\", "\\\\", Object, fixed=TRUE)) %>% # escape backslashes :(
+  write_delim("data/ADP.nq", col_names=FALSE, escape="none", quote="none")
+
+unescape_unicode <- function(x) {
+  stringi::stri_unescape_unicode(gsub("<U\\+(....)>", "\\\\u\\1", x))
+}
+
+# prove this parses correctly
+library(rdflib)
+true_triples <- read_nquads("data/ADP.nq")
+
+# serialize to any format
+rdflib::rdf_serialize(true_triples, "data/ADP.ttl",
+                      namespace = c(APD = "https://github.com/traitecoevo/APD/",
+                                    dc = "http://purl.org/dc/elements/1.1/",
+                                    skos = "http://www.w3.org/2004/02/skos/core#",
+                                    dwc = "http://rs.tdwg.org/dwc/terms/attributes/",
+                                    dcam = "http://purl.org/dc/dcam/",
+                                    dcterms = "http://purl.org/dc/terms/",
+                                    ets = "http://terminologies.gfbio.org/terms/ETS/",
+                                    obo = "http://purl.obolibrary.org/obo/",
+                                    oboecore = "http://ecoinformatics.org/oboe/oboe.1.2/oboe-core.owl#",
+                                    ont = "https://w3id.org/iadopt/ont/",
+                                    owl = "http://www.w3.org/2002/07/owl#",
+                                    rdfs = "http://www.w3.org/2000/01/rdf-schema#",
+                                    uom = "https://w3id.org/uom/",
+                                    datacite = "http://purl.org/datacite/v4.4/",
+                                    xsd = "http://www.w3.org/2001/XMLSchema#",
+                                    Cerrado = "http://cerrado.linkeddata.es/ecology/",
+                                    CorVeg  = "http://linked.data.gov.au/def/corveg-cv/",
+                                    DCM = "http://dicom.nema.org/resources/ontology/DCM/",
+                                    EDAM = "http://edamontology.org/ ",
```

Comment:
whoops, note the extraneous space included in the EDAM URL, this will create problems.